manticore/wasm/structure.py

```diff
@@ -0,0 +1,1703 @@
+            assert isinstance(values[idx], global_i.type.value)
+            self.globaladdrs.append(global_i.allocate(store, values[idx]))
+        for idx, export_i in enumerate(module.exports):
+            if isinstance(export_i.desc, FuncIdx):
+                self.exports.append(ExportInst(export_i.name, self.funcaddrs[export_i.desc]))
+            elif isinstance(export_i.desc, TableIdx):
+                self.exports.append(ExportInst(export_i.name, self.tableaddrs[export_i.desc]))
+            elif isinstance(export_i.desc, MemIdx):
+                self.exports.append(ExportInst(export_i.name, self.memaddrs[export_i.desc]))
+            elif isinstance(export_i.desc, GlobalIdx):
+                self.exports.append(ExportInst(export_i.name, self.globaladdrs[export_i.desc]))
+            else:
+                raise RuntimeError("Export desc wasn't a function, table, memory, or global")
+            self.export_map[export_i.name] = len(self.exports) - 1
+
+    def invoke_by_name(self, name: str, stack, store, argv):
+        """
+        Iterates over the exports, attempts to find the function specified by `name`. Calls `invoke` with its FuncAddr,
+        passing argv
+
+        :param name: Name of the function to look for
+        :param argv: Arguments to pass to the function. Can be BitVecs or Values
+        """
+        for export in self.exports:
+            if export.name == name and isinstance(export.value, FuncAddr):
+                return self.invoke(stack, export.value, store, argv)
+        raise RuntimeError("Can't find a function called", name)
+
+    def invoke(self, stack: "Stack", funcaddr: FuncAddr, store: Store, argv: typing.List[Value]):
+        """
+        Invocation wrapper. Checks the function type, pushes the args to the stack, and calls _invoke_inner.
+        Unclear why the spec separates the two procedures, but I've tried to implement it as close to verbatim
+        as possible.
+
+        Note that this doesn't actually _run_ any code. It just sets up the instruction queue so that when you call
+        `exec_instruction, it'll actually have instructions to execute.
+
+        https://www.w3.org/TR/wasm-core-1/#invocation%E2%91%A1
+
+        :param funcaddr: Address (in Store) of the function to call
+        :param argv: Arguments to pass to the function. Can be BitVecs or Values
+        """
+        assert funcaddr in range(len(store.funcs))
+        funcinst = store.funcs[funcaddr]
+        ty = funcinst.type
+        assert len(ty.param_types) == len(
+            argv
+        ), f"Function {funcaddr} expects {len(ty.param_types)} arguments"
+        # for t, v in zip(ty.param_types, argv):
+        #     assert type(v) == type(t)
+
+        # Create dummy frame, which is required by the spec but doesn't serve any obvious purpose. It also never
+        # gets popped.
+        dummy_frame = Frame([], ModuleInstance())
+        stack.push(dummy_frame)  # type: ignore
```

Comment:
Due to blackening I typically wasn't able to put it on the same line, but I've added annotating comments on the previous line from each `type: ignore`.

It may become problematic to have mypy, which partially relies on comments, fighting with `black`, which treats comments as unimportant parts of the AST it can move to wherever.

Reply:
:( Okay, yeah a comment above is just as good! Thank you!!